frontend/src/routes/CriticalQuestions/index.js

```diff
@@ -539,9 +560,33 @@ function CriticalQuestions(props) {
         <div style={{ height: "30px" }} ref={pageBottomRef}></div>
       </div>
 
+      {(contacts.filter((contact) => contact.email).length === 0 ||
+        !formValues[fields.CITY.key] ||
+        !formValues[fields.CITY.key].length ||
+        !formValues[fields.STATE.key] ||
+        !formValues[fields.STATE.key].length ||
+        !formValues[fields.COUNTRY.key] ||
+        !formValues[fields.COUNTRY.key].length) && (
+        <Fab
+          style={{ background: "#EA2027" }}
+          aria-label="Go to next page"
+          size="medium"
+          className="fab next-btn"
+          onClick={handleSubmit}
+        >
+          <ArrowRightIcon />
+        </Fab>
+      )}
+
       {contacts.filter((contact) => contact.email).length !== 0 &&
-        formValues[fields.CITY.key] &&
-        formValues[fields.CITY.key].length &&
+        ((next_count === 0 &&
+          !!(
```

Comment:
double ! ?

Reply:
Yep. With no `!` it actually returns an `<empty string>`, I tested it so I did this instead.

Reply:
Then, why it is not !! on line 591. If it returns empty string, then it fits the condition `fromValues[fileds.CITY.key].length`. It's duplicate.

Reply:
Basically `!` makes the value boolean, and I used one `!` to check if it's false on line 591 (around). And with `!!` I want to check if it's true. I am not sure what do you mean in the second sentence about duplicates.

Reply:
Before I used `formValues[fields.CITY.key]` to check if it null(it is possible that it's null). If you just want to check if it empty string, then `formValues[fields.CITY.key].length` already checks. Also, instead of using `!!` just use `=== ""`

Reply:
Hmm the thing is I want to check if it's null, and I am basically just following your logic, but it somehow does not render correctly to evaluate to boolean.

Reply:
I am not actually sure why it evaluate to `<emptry string>` when I log it to console, but I don't actually want to check for `=== ""` because that's not what's being checked, but that's somehow what it evaluate to.

Reply:
!! means nothing like it is still the original boolean.

Reply:
Please print the out and see what I am saying.

Reply:
I deleted !!. It behaves the same. ! does not change the string or object into boolean. It's if it is not null, then it's true. Then, the first ! makes it false and the second ! makes it true again.

Reply:
Like it actually mysteriously evaluate to `<empty string>` even though it shouldn't and it throws the entire thing off. And doing `!` would turn it to a boolean and be evaluated properly.

Reply:
Before you use ! it's already a boolean. Otherwise, you cannot use ! to turn it into a boolean.

Reply:
If there is something wrong with it, what would happen if `!!` is deleted?

Reply:
I think it actually doesn't work. Like If you log `fromValues[fileds.CITY.key]&& fromValues[fileds.CITY.key].length `, it would give you `<empty string>` and `!` it would mysteriously turn it to `false`. I am not actually sure why. And it's what I observed when I wrote it. Please feel free to test it and remove the `!!` if it works.

Reply:
I c.